packages/driver/src/cypress/error_messages.ts

```diff
@@ -1880,26 +1876,49 @@ export default {
 
         Cypress only considers the \`window\`, \`document\`, or any \`element\` to be valid DOM objects.`
     },
-    not_attached (obj) {
+    detached_during_actionability (obj) {
       return {
         message: stripIndent`\
-          ${cmd(obj.cmd)} failed because this element is detached from the DOM.
+          ${cmd(obj.name)} failed because the page updated while this command was executing. Cypress tried to locate elements based on this query:
 
-          \`${obj.node}\`
+          > ${subjectChainToString(obj.subjectChain)}
+
+          We initially found matching element(s), but while waiting for them to become actionable, they disappeared from the page. Common situations why this happens:
+            - Your JS framework re-rendered asynchronously
+            - Your app code reacted to an event firing and removed the element
 
-          Cypress requires elements be attached in the DOM to interact with them.
+          You can typically solve this by breaking up a chain. For example, rewrite:
 
-          The previous command that ran was:
+          > \`cy.get('button').click().click()\`
 
-            > ${cmd(obj.prev)}
+          to
 
-          This DOM element likely became detached somewhere between the previous and current command.
+          > \`cy.get('button').click()\`
+          > \`cy.get('button').click()\`
+
+            `,
+        docsUrl: 'https://on.cypress.io/element-has-detached-from-dom',
+      }
+    },
+    detached_after_command (obj) {
+      return {
+        message: stripIndent`\
+          ${cmd(obj.name)} failed because the page updated as a result of this command, but you tried to continue the command chain. The subject is no longer attached to the DOM, and Cypress cannot requery the page after commands such as ${cmd(obj.name)}.
```

Comment:
This wording standalone is a little confusing. Wouldn't the error be on the continued command chain when the dom is no longer on the page?
```suggestion
          ${cmd(obj.name)} failed because the page mutated after running the previous command. The subject is no longer attached to the DOM, and Cypress cannot requery the page to run ${cmd(obj.name)}.
```

Reply:
I'm going to disagree here: I think the existing wording more accurately reflects the actual problem.

I click a link, the page navigates.

"The page updated as a result of this command" is clearer than "The page mutated after running this command."

Reply:
I'm confused, Is this failure associated with a command retry or a chained child command?

Reply:
This is associated with a failure during verifyUpcomingAssertions.

You do something like:

```
cy.get('button').click().should('have.class', 'active')
```

and the element gets detached *after clicking*, but while verifying `.should()`.